### PR TITLE
Filter out unauthored lists on UserListsPage and AddToListDialog

### LINKED ISSUE #2382: Favorited user lists leak into UI that shows my own lists
### Steps to Reproduce

- login as user A, confirm you have no lists at /courses/lists/
- as user A, favorite some of user B's lists
- as user A navigate directly to /courses/search (so your react state starts from zero)
- click on the star next to a search result and observe that these lists from the other user that you favorited appear in your lists
- click the "My Lists" link on the header and observe that the lists you favorited show up as yours

The issue seems to be that this code transforms those objects onto `entities.userLists`, but the UI pieces don't filter those down to lists authored by the current user:
https://github.com/mitodl/open-discussions/blob/master/static/js/lib/queries/learning_resources.js#L36-L47

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #2382 

#### What's this PR do?
Filters out any lists in `state.entities.userLists` that are not authored by the current user

#### How should this be manually tested?
- As user A, search for user lists or learning paths and favorite 1+ that are authored by user B
- Go to the "My Lists" page, the recently favorited lists by user B should not show up here
- Click the star on any resource card to open the 'Add to List' dialog, only your own lists should show up here as well.

### CODE REVIEW

**Comment by aliceriot** [comment]
works great! looking at code now.

File: static/js/factories/learning_resources.js [hunk 1]
```diff
@@ -163,7 +163,8 @@ export const makeUserList = (): UserList => ({
   list_type:     "userlist",
   profile_img:   casual.url,
   profile_name:  casual.name,
-  privacy_level: casual.random_element(["public", "private"])
+  privacy_level: casual.random_element(["public", "private"]),
+  author:        casual.number
```
Comment (aliceriot) [inline static/js/factories/learning_resources.js]: I think here for `author` you need to use `casual.integer(low, high)`, where `low` and `high` are integer values for a range from which the function should select an integer (e.g. 1 and 1000 or something)

Comment (aliceriot) [reply static/js/factories/learning_resources.js]: just realized we are using `casual.number` in a number of places - but if you check in the repl `casual.number` isn't a function that `casual` provides

Comment (mbertrand) [reply static/js/factories/learning_resources.js]: I replaced all the `casual.number`s with `casual.integer`s

**Pull request merged**